qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -313,6 +322,217 @@ def _run(self, qobj, job_id=''):
+                           [q2i[q] for q in qargs],
+                           [c2i[c] for c in cargs])
+            elif isinstance(inst, WhileLoopOp):
+                (cond_tuple, continue_label,
+                 loop_start_label, break_label, body) = self._inline_while_loop_op(inst)
+                c_if_args = self._convert_c_if_args(cond_tuple)
+                ret.append(AerMark(continue_label, ret.num_qubits),
+                           range(ret.num_qubits), [])
+                ret.append(AerJump(loop_start_label, ret.num_qubits).c_if(*c_if_args),
+                           range(ret.num_qubits), [])
+                ret.append(AerJump(break_label, ret.num_qubits),
+                           range(ret.num_qubits), [])
+                ret.append(AerMark(loop_start_label, ret.num_qubits),
+                           range(ret.num_qubits), [])
+                ret.append(body, [q2i[q] for q in qargs], [c2i[c] for c in cargs])
+            elif isinstance(inst, IfElseOp):
+                cond_tuple, true_label, else_label, body = self._inline_if_else_op(inst,
+                                                                                   continue_label,
+                                                                                   break_label)
+                c_if_args = self._convert_c_if_args(cond_tuple)
+                ret.append(AerJump(true_label, ret.num_qubits).c_if(*c_if_args),
+                           range(ret.num_qubits), [])
+                ret.append(AerJump(else_label, ret.num_qubits),
+                           range(ret.num_qubits), [])
+                ret.append(body, [q2i[q] for q in qargs], [c2i[c] for c in cargs])
+            elif isinstance(inst, BreakLoopOp):
+                ret.append(AerJump(break_label, ret.num_qubits),
+                           range(ret.num_qubits), [])
+            elif isinstance(inst, ContinueLoopOp):
+                ret.append(AerJump(continue_label, ret.num_qubits),
+                           range(ret.num_qubits), [])
+            else:
+                ret.append(inst, qargs, cargs)
+
+        return ret
+
+    def _compile(self, circuits):
+        if isinstance(circuits, list):
+            all_static = all([not self._is_dynamic(circuit) for circuit in circuits])
+            if all_static:
+                return circuits
+            basis_gates = self.configuration().basis_gates + ['mark', 'jump']
+            return [transpile(self._inline_circuit(self._decompose_subcircuits(circuit),
+                                                   None, None),
+                              basis_gates=basis_gates) if self._is_dynamic(circuit)
+                    else circuit for circuit in circuits]
+        else:
+            return (transpile(self._inline_circuit(self._decompose_subcircuits(circuits),
+                                                   None, None),
+                              basis_gates=basis_gates) if self._is_dynamic(circuits)
```

Comment:
`basis_gates` is not defined here, since it is only defined in the `if` block of this conditional